prometheus_client/context_managers.py

```diff
@@ -42,19 +42,26 @@ def wrapped(func, *args, **kwargs):
 
 
 class Timer:
-    def __init__(self, callback):
-        self._callback = callback
+    def __init__(self, metric, callback_name):
+        self._metric = metric
+        self._callback_name = callback_name
 
     def _new_timer(self):
-        return self.__class__(self._callback)
+        return self.__class__(self._metric, self._callback_name)
 
     def __enter__(self):
         self._start = default_timer()
+        return self
 
     def __exit__(self, typ, value, traceback):
         # Time can go backwards.
         duration = max(default_timer() - self._start, 0)
-        self._callback(duration)
+        self._metric._raise_if_not_observable()
```

Comment:
I actually don't think this is necessary as `set` or `observe` will already call `_raise_if_not_observable`.

Reply:
Good point, I was trying to "make up" [for the call on the way in](https://github.com/prometheus/client_python/pull/730/files#diff-c146acd4b81d5b8995b2567ff78b9c11df02c051971f7e66f44bc6f765031273L399), essentially making sure that it's still called at all, but I didn't realize that the callbacks already do so anyway… 🙂